cmd/client-admin.go

```diff
@@ -57,9 +57,7 @@ func NewAdminFactory() func(config *Config) (*madmin.AdminClient, *probe.Error)
 		var found bool
 		if api, found = clientCache[confSum]; !found {
 
-			transport := getTransportForConfig(config, true)
-
-			credsChain, err := getCredentialsChainForConfig(config, transport)
+			transport, credsChain, err := getTransportAndCredentialsChainForConfig(config)
```

Comment:
For a cleaner implementation move this as a method for the `config` such as 

```
config.getTransport()
```

```
config.getCredsChain()
```

Just more cleaner separation of functions.